common/schema-model/src/main/java/io/smallrye/graphql/schema/model/Reference.java

```diff
@@ -134,6 +138,29 @@ public boolean isAdaptingWith() {
         return this.adaptWith != null;
     }
 
+    public Map<String, Reference> getAllParametrizedTypes() {
+        if (classParametrizedTypes == null) {
+            return extendedClassParametrizedTypes; // if both are 'null', then it returns null
```

Comment:
Why should return null here, how about we just combine the two maps, possibly resulting in an empty map?
Neither of the two maps should be null, we should make sure both are at least empty.